File: go/mission/evidence.go
package mission

// P7A — the Evidence Plane: an append-only kernel primitive over the mission event store. Evidence,
// claims and verification results are first-class because verification, explanation, approvals,
// replay and learning all read from here. Go port of agentic_os/mission/evidence.py.

type EvidenceLog struct{ store *EventStore }

func NewEvidenceLog(store *EventStore) *EvidenceLog { return &EvidenceLog{store} }

func (l *EvidenceLog) RaiseClaim(missionID string, claim Claim) Claim {
	l.store.Append("ClaimRaised", missionID, jsonableMap(claim))
	return claim
}

func (l *EvidenceLog) RecordEvidence(missionID string, ev EvidenceRecord) EvidenceRecord {
	l.store.Append("EvidenceRecorded", missionID, jsonableMap(ev))
	return ev
}

func (l *EvidenceLog) RecordVerification(missionID, nodeID string, vr VerificationResult) {
	p := jsonableMap(vr)
	p["node_id"] = nodeID
	l.store.Append("VerificationRecorded", missionID, p)
}

// Ledger returns the full evidence ledger for a mission (claims · evidence · verifications), in order.
func (l *EvidenceLog) Ledger(missionID string) []map[string]any {
	kinds := map[string]bool{"ClaimRaised": true, "EvidenceRecorded": true, "VerificationRecorded": true}
	var out []map[string]any
	for _, e := range l.store.ForMission(missionID) {
		if kinds[e.Type] {
			m := map[string]any{"type": e.Type}
			for k, v := range e.Payload {
				m[k] = v
			}
			out = append(out, m)
		}
	}
	return out
}

File: agentic_os/mission/evidence.py
"""P7A — the Evidence Plane: an append-only kernel primitive over the mission event store.

Evidence, claims and verification results are first-class because verification, explanation,
approvals, replay and learning all read from here — not because a UI needs them. Every record is
an event, so the ledger is durable, replayable and audit-grade for free.
"""
from __future__ import annotations

from .types import Claim, EvidenceRecord, VerificationResult, to_jsonable


class EvidenceLog:
    def __init__(self, store):
        self.store = store

    def raise_claim(self, mission_id: str, claim: Claim) -> Claim:
        self.store.append("ClaimRaised", mission_id, to_jsonable(claim))
        return claim

    def record_evidence(self, mission_id: str, ev: EvidenceRecord) -> EvidenceRecord:
        self.store.append("EvidenceRecorded", mission_id, to_jsonable(ev))
        return ev

    def record_verification(self, mission_id: str, node_id: str, vr: VerificationResult) -> None:
        self.store.append("VerificationRecorded", mission_id,
                          {"node_id": node_id, **to_jsonable(vr)})

    def ledger(self, mission_id: str) -> list[dict]:
        """The full evidence ledger for a mission (claims · evidence · verifications), in order."""
        kinds = {"ClaimRaised", "EvidenceRecorded", "VerificationRecorded"}
        return [{"type": e.type, **e.payload} for e in self.store.for_mission(mission_id)
                if e.type in kinds]
